core/src/test/java/org/apache/iceberg/TestCatalogUtil.java

```diff
@@ -123,6 +123,16 @@ public void loadCustomFileIO_noArg() {
     Assert.assertEquals(properties, ((TestFileIONoArg) fileIO).map);
   }
 
+  @Test
+  public void loadCustomFileIO_hadoopConfigConstructor() {
+    Configuration configuration = new Configuration();
+    configuration.set("key", "val");
+    FileIO fileIO = CatalogUtil.loadFileIO(
+        TestFileIOHadoopConstructor.class.getName(), Maps.newHashMap(), configuration);
```

Comment:
Why not just  load the `HadoopFileIO` directly ?  Here we are trying to load an `TestFileIOHadoopConstructor`,  I'm concerning that we could not detect the breaking case if someone change the `HadoopFileIO` constructor....

Reply:
I don't think that would be possible, since HadoopFileIO is public and the constructor is also public, we cannot simply change it. But this is a good point, I can just load it instead of having this dummy test class.